sql/core/src/test/resources/sql-tests/analyzer-results/udf/postgreSQL/udf-select_having.sql.out

```diff
@@ -102,12 +102,11 @@ Project [udf(b)#x, udf(c)#x]
 SELECT udf(b), udf(c) FROM test_having
 	GROUP BY b, c HAVING udf(b) = 3 ORDER BY udf(b), udf(c)
 -- !query analysis
-Project [udf(b)#x, udf(c)#x]
```

Comment:
do you know why the plan is changed?

Reply:
Though this query has both `HAVING` and `ORDER BY` clauses, but only scalar function is present in `ORDER BY` clause, I think the previous resolution matches item 4 of `ResolveReferencesInSort` comments.

> 4. Resolves the column to [[AttributeReference]] with the output of a descendant plan node.
>    Spark will propagate the missing attributes from the descendant plan node to the Sort node.
>    This is to allow users to ORDER BY columns that are not in the SELECT clause, which is
>    widely supported in other SQL dialects. For example, `SELECT a FROM t ORDER BY b`.

With this patch, it should match item 3

> 3. If the child plan is Aggregate or Filter(_, Aggregate), resolves the column to
>    [[TempResolvedColumn]] with the output of Aggregate's child plan.
>    This is to allow Sort to host grouping expressions and aggregate functions, which can
>    be pushed down to the Aggregate later. For example,
>    `SELECT max(a) FROM t GROUP BY b HAVING max(a) > 1 ORDER BY min(a)`.

Reply:
oh so the plan is actually more efficient?

Reply:
i think so, the plan shows it eliminates some unnecessary column propagation across operators

Reply:
it is an analyzed plan, the optimized plan should be same with pr ?

Reply:
both Analyzed plan and Optimized plan are changed :)

before:
```
== Analyzed Logical Plan ==
udf(b): int, udf(c): double
Project [udf(b)#24, udf(c)#25]
+- Sort [udf(b#21) ASC NULLS FIRST, udf(cast(c#22 as double)) ASC NULLS FIRST], true
   +- Filter (udf(b)#24 = 3)
      +- Aggregate [b#21, c#22], [udf(b#21) AS udf(b)#24, udf(cast(c#22 as double)) AS udf(c)#25, b#21, c#22]
         +- SubqueryAlias spark_catalog.default.test_having
            +- Relation spark_catalog.default.test_having[a#20,b#21,c#22,d#23] parquet

== Optimized Logical Plan ==
Project [udf(b)#24, udf(c)#25]
+- Sort [udf(b#21) ASC NULLS FIRST, udf(cast(c#22 as double)) ASC NULLS FIRST], true
   +- Aggregate [b#21, c#22], [udf(b#21) AS udf(b)#24, udf(cast(c#22 as double)) AS udf(c)#25, b#21, c#22]
      +- Project [b#21, c#22]
         +- Filter (isnotnull(b#21) AND (udf(b#21) = 3))
            +- Relation spark_catalog.default.test_having[a#20,b#21,c#22,d#23] parquet
```

after:
```
== Analyzed Logical Plan ==
udf(b): int, udf(c): double
Sort [udf(b)#9 ASC NULLS FIRST, udf(c)#10 ASC NULLS FIRST], true
+- Filter (udf(b)#9 = 3)
   +- Aggregate [b#6, c#7], [udf(b#6) AS udf(b)#9, udf(cast(c#7 as double)) AS udf(c)#10]
      +- SubqueryAlias spark_catalog.default.test_having
         +- Relation spark_catalog.default.test_having[a#5,b#6,c#7,d#8] parquet

== Optimized Logical Plan ==
Sort [udf(b)#9 ASC NULLS FIRST, udf(c)#10 ASC NULLS FIRST], true
+- Aggregate [b#6, c#7], [udf(b#6) AS udf(b)#9, udf(cast(c#7 as double)) AS udf(c)#10]
   +- Project [b#6, c#7]
      +- Filter (isnotnull(b#6) AND (udf(b#6) = 3))
         +- Relation spark_catalog.default.test_having[a#5,b#6,c#7,d#8] parquet
```